src/ucp/wireup/wireup_cm.c

```diff
@@ -1374,7 +1374,7 @@ void ucp_ep_cm_disconnect_cm_lane(ucp_ep_h ucp_ep)
     ucs_assert_always(uct_cm_ep != NULL);
     /* No reason to try disconnect twice */
     ucs_assert(!(ucp_ep->flags & UCP_EP_FLAG_DISCONNECTED_CM_LANE));
-    ucs_assert(!(ucp_ep->flags & UCP_EP_FLAG_FAILED));
+    ucs_assert(ucp_ep_is_cm_local_connected(ucp_ep));
```

Comment:
isn't it a duplication of previous assert?
btw, do we really need `UCP_EP_FLAG_DISCONNECTED_CM_LANE`?

Reply:
> isn't it a duplication of previous assert?

no, we just need to check this function is called only if CM is connected

> btw, do we really need `UCP_EP_FLAG_DISCONNECTED_CM_LANE`?

yes, we use it for assertions in some places

Reply:
but we also check `ucs_assert(!(ucp_ep->flags & UCP_EP_FLAG_DISCONNECTED_CM_LANE))`

Reply:
yes, but we check whether CM is connected or not to make sure that we call CM disconnect only when it is true